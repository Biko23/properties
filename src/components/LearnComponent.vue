<template>
  <div>
    <v-container id="container" fluid>
      <p id="intro">With us you can sell your home with no fear</p>
      <p id="intro2">Directly to us, through an agent, or yourself</p>
    </v-container>
    <br />
    <div style="text-align: center; margin-top: 15px">
      <h3><span style="color: #3b6ef3"> LET'S </span>GUIDE YOU</h3>
    </div>
    <v-container>
      <v-row>
        <v-col></v-col>
        <v-col cols="12" md="5" style="background-color: #f2f2f2">
          <div>
            <v-card-title>Sell your home directly to us </v-card-title>

            <v-card-subtitle>
              Getting a competitive cash offer from Stanbic Properties is an
              easy path to sell your home, with more control and less prepwork.
              You sell your home as-is and get paid with no surprises, while we
              take care of the rest.
              <h4>Ask how much we can pay</h4>
              <br />
              <v-row>
                <v-col cols="12" sm="12">
                  <v-form ref="form" v-model="valid" lazy-validation>
                    <v-text-field
                      v-model="name"
                      :rules="nameRules"
                      label="From"
                      required
                      solo
                    ></v-text-field>

                    <v-text-field
                      v-model="email"
                      :rules="emailRules"
                      label="To"
                      required
                      solo
                    ></v-text-field>

                    <v-text-field
                      v-model="email"
                      :rules="emailRules"
                      label="Property Details"
                      required
                      solo
                    ></v-text-field>
                    <v-btn color="primary" class="mr-4" @click="reset">
                      Submit
                    </v-btn>
                  </v-form>
                </v-col>
              </v-row>
            </v-card-subtitle>
          </div>
        </v-col>
        <v-col cols="12" md="5">
          <v-card max-width="auto">
            <v-img
              src="../assets/finishhouse.jpg"
              height="500px"
            ></v-img>
          </v-card>
        </v-col>
        <v-col></v-col> </v-row
      ><br /><br />

      <v-row>
        <v-col></v-col>
        <v-col cols="12" md="5">
          <v-card max-width="auto">
            <v-img
              src="../assets/broker.jpg"
              height="300px"
            ></v-img>
          </v-card>
        </v-col>
        <v-col cols="12" md="5" style="background-color: #f2f2f2">
          <div>
            <v-card-title> Sell your home via a Broker </v-card-title>

            <v-card-subtitle style="margin:auto; padding:10px;">
              When you work with a real estate agent, you'll get selling support
              at every step, from prepping and listing your home to marketing
              that gets buyers in the door.

              <p style="text-align: center; margin-top: 1rem">
                <v-btn style="background-color: #3b6ef3">
                  <span
                    style="
                      color: #ffffff;
                      font-size: 20px;
                      font-style: normal;
                      font-weight: 300;
                      line-height: 30px;
                      letter-spacing: 0em;
                      text-align: center;
                      text-transform: capitalize;
                    "
                  >
                    Get a Broker</span
                  >
                </v-btn>
              </p>
            </v-card-subtitle>
          </div>
        </v-col>
        <v-col></v-col> </v-row
      ><br /><br />

      <v-row
        ><br />
        <v-col></v-col>
        <v-col cols="12" md="5" style="background-color: #f2f2f2">
          <div>
            <v-card-title> Sell your home yourself</v-card-title>

            <v-card-subtitle style="margin:auto; padding:10px;">
              Sell your home yourself Deciding to sell your home yourself is
              referred to as for-sale-by-owner (FSBO). The FSBO process is
              similar to traditional selling, but without the help of a real
              estate agent. In this case, youâ€™re responsible for the home prep,
              marketing, showings, and negotiations.
              <p style="text-align: center; ">
                <v-btn style="background-color: #3b6ef3;  width:100px">
                  <span
                    style="
                      color: #ffffff;
                      font-size: 20px;
                      font-style: normal;
                      font-weight: 300;
                      line-height: 30px;
                      letter-spacing: 0em;
                      text-align: center;
                      text-transform: capitalize;
                    "
                  >
                    Sell
                  </span>
                </v-btn>
              </p>
            </v-card-subtitle>
          </div>
        </v-col>
        <v-col cols="12" md="5">
          <v-card max-width="auto">
            <v-img
              src="../assets/landlord.jpg"
              height="300px"
            ></v-img>
          </v-card>
        </v-col>
        <v-col></v-col>
      </v-row>
    </v-container><br>
    <Footer />
  </div>
</template>
<script>
import Footer from "./Footer.vue";
export default {
  components: { Footer },
  name: "LearnComponent",
};
</script>
<style scoped>
#intro {
  color: #ffffff;
  font-size: 1.5rem;
  font-weight: 400;
  text-align: center;
  margin-top: 180px;
}
#intro2 {
  color: #ffffff;
  font-size: 1.2rem;
  font-weight: 400;
  text-align: center;
}
#card {
  background-color: #f4f5f8;
}
#container {
  background-image: url("https://res.cloudinary.com/diued7ugb/image/upload/v1626688778/dillon-kydd-XGvwt544g8k-unsplash_lrvzid.jpg");
  background-size: cover;
  height: 75vh;
  width: 100%;
}

@media only screen and (max-width: 768px) {
  #container {
    height: 30vh;
    width: 100%;
  }
  #intro {
    color: #ffffff;
    font-size: 1rem;
    font-weight: 400;
    text-align: center;
    margin-top: auto;
  }
  #intro2 {
    color: #ffffff;
    font-size: 0.8rem;
    font-weight: 400;
    text-align: center;
  }
  #col-hiden {
    display: none;
  }
}
</style>