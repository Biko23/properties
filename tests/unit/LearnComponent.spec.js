import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import LearnComponent from '@/components/LearnComponent.vue'

describe('LearnComponent.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let vuetify
    let actions
    let store
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            postAUserLog: jest.fn()
        }
        store = new Vuex.Store({
            actions
        })
    })

    const mountFunction = options => {
        return mount(LearnComponent, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFunction()
        expect(wrapper.html()).toMatchSnapshot()
    })
})

