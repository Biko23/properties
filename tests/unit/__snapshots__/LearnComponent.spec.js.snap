// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnComponent.vue should have a match snapshot 1`] = `
<div>
  <div class="container container--fluid" id="container">
    <p id="intro">With us you can sell your home with no fear</p>
    <p id="intro2">Directly to us, through an agent, or yourself</p>
  </div> <br>
  <div style="text-align: center; margin-top: 15px;">
    <h3><span style="color: rgb(59, 110, 243);"> LET'S </span>GUIDE YOU</h3>
  </div>
  <div class="container">
    <div class="row">
      <div class="col"></div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title">Sell your home directly to us </div>
          <div class="v-card__subtitle">
            Getting a competitive cash offer from Stanbic Properties is an
            easy path to sell your home, with more control and less prepwork.
            You sell your home as-is and get paid with no surprises, while we
            take care of the rest.
            <h4>Ask how much we can pay</h4> <br>
            <div class="row">
              <div class="col-sm-12 col-12">
                <form novalidate="novalidate" class="v-form">
                  <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">From</label><input required="required" id="input-5" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-8" class="v-label theme--light" style="left: 0px; position: absolute;">To</label><input required="required" id="input-8" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label for="input-11" class="v-label theme--light" style="left: 0px; position: absolute;">Property Details</label><input required="required" id="input-11" type="text"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                        </div>
                      </div>
                    </div>
                  </div> <button type="button" class="mr-4 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">
                    Submit
                  </span></button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-md-5 col-12">
        <div class="v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 500px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
      <div class="col"></div>
    </div><br><br>
    <div class="row">
      <div class="col"></div>
      <div class="col-md-5 col-12">
        <div class="v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 300px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title"> Sell your home via a Broker </div>
          <div class="v-card__subtitle" style="margin: auto; padding: 10px;">
            When you work with a real estate agent, you'll get selling support
            at every step, from prepping and listing your home to marketing
            that gets buyers in the door.

            <p style="text-align: center; margin-top: 1rem;"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="background-color: rgb(59, 110, 243);"><span class="v-btn__content"><span style="color: rgb(255, 255, 255); font-size: 20px; font-style: normal; font-weight: 300; line-height: 30px; letter-spacing: 0em; text-align: center; text-transform: capitalize;">
                  Get a Broker</span></span></button></p>
          </div>
        </div>
      </div>
      <div class="col"></div>
    </div><br><br>
    <div class="row"><br>
      <div class="col"></div>
      <div class="col-md-5 col-12" style="background-color: rgb(242, 242, 242);">
        <div>
          <div class="v-card__title"> Sell your home yourself</div>
          <div class="v-card__subtitle" style="margin: auto; padding: 10px;">
            Sell your home yourself Deciding to sell your home yourself is
            referred to as for-sale-by-owner (FSBO). The FSBO process is
            similar to traditional selling, but without the help of a real
            estate agent. In this case, youâ€™re responsible for the home prep,
            marketing, showings, and negotiations.
            <p style="text-align: center;"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default" style="background-color: rgb(59, 110, 243); width: 100px;"><span class="v-btn__content"><span style="color: rgb(255, 255, 255); font-size: 20px; font-style: normal; font-weight: 300; line-height: 30px; letter-spacing: 0em; text-align: center; text-transform: capitalize;">
                  Sell
                </span></span></button></p>
          </div>
        </div>
      </div>
      <div class="col-md-5 col-12">
        <div class="v-card v-sheet theme--light" style="max-width: auto;">
          <div class="v-image v-responsive theme--light" style="height: 300px;">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
      <div class="col"></div>
    </div>
  </div>
</div>
`;
